src/main/java/com/redfin/insist/impl/AbstractFutureImpl.java

```diff
@@ -65,4 +68,14 @@
             }
         };
     }
+
+    @Override
+    public void thatEventuallyIsPresent(Supplier<Optional<?>> supplier) throws X {
+        this.thatEventually(() -> supplier.get().isPresent());
```

Comment:
In both this, and the following method, it will throw a NullPointerException if supplier is null rather than an IllegalArgumentException that it should throw. Need to add as the first line for both methods (before calling `this.thatEventually`:
validate().that(supplier).isNotNull();

Reply:
Also, for the `Supplier<Optional<?>>` one, the optional itself should be checked for non-null as well to avoid NullPointerException:
```
validate().that(supplier).isNotNull();
this.thatEventually(() ->{
    Optional<?> optional = supplier.get();
    return null != optional && optional.isPresent();
});
```

Reply:
Ah.  Good catch on the supplier.  I will add the additional null check as well, though people shouldn't be returning null for an optional.